### PR TITLE
Addon-docs: Dynamic Source rendering for React

### PR DESCRIPTION
**Pull request by shilman**
Issue: #8342 #8444 

## What I did

dyanamic-source

Work in progress for rendering JSX in the code snippet, based on  https://github.com/storybookjs/addon-jsx (cc @hipstersmoothie). Now, for React (and eventually all view layers), the JSX of the story function output is rendered to an example snippet dynamically.

This solves the following issues:
- [x] Show controls/knobs values dynamically #8342 
- [x] Fix imported story functions from CSF into MDX #8444 
- [x] Fix `Story.bind({})` source for args stories

Still to be done:
- [ ] Expanded test cases
  - [ ] Handle large objects/arrays gracefully
  - [ ] Handle corner cases like `memo` and `ref`s
- [ ] Figure out source customization
- [ ] Figure out framework compatibility path forward

## What I need

- [ ] Feedback on the approach and test cases we need to worry about to make this robust
- [ ] Volunteers to help port this to non-React frameworks

## How to test

See `official-storybook`

### CODE REVIEW

File: addons/docs/src/frameworks/common/preset.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export function webpack(webpackConfig: any = {}, options: any = {}) {
   const {
     babelOptions,
     configureJSX = options.framework !== 'react', // if not user-specified
-    sourceLoaderOptions = {},
+    sourceLoaderOptions = options.framework === 'react' || {},
```
Comment (tmeasday) [inline addons/docs/src/frameworks/common/preset.ts:44]: What does this mean?

Comment (shilman) [reply addons/docs/src/frameworks/common/preset.ts]: fixed

**Comment by shilman** [comment]
Merging this and releasing so that React users can start trying it out to keep things moving.

**Comment by ndelangen** [comment]
I love this, and would love it even more, if you could switch to the HTML output as well somehow!

❤️ ♥️ ❤️

**Comment by maggiewachs** [comment]
@shilman is there an open issue that's tracking progress for non-React frameworks? Specifically, I'm looking for the code block to work with the web components build. (Many thanks for all of your work on this project! We're really digging it at Filament Group so far.)

**Comment by shilman** [comment]
Thanks @maggiewachs! There is now 😅 

https://github.com/storybookjs/storybook/issues/11588

I had somebody else ask about lit-element support yesterday & volunteer to contribute, so hopefully we can get that going in 6.1

**Comment by ecarrera** [comment]
It will be really useful @shilman, great!!  ❤️ I was thinking about it today.

**Pull request merged**